### ISSUE TITLE
Multiple messages when using chatAddPattern

### ISSUE THREAD

**Issue by GoogleSites** [PrismarineJS/mineflayer#795]
If you are reporting a bug please report on these points :

* what you want to do (the objective in english)
- Show the output of the message once.
* what you tried (the code)
https://pastebin.com/VWFxHF0M

* what happened
On the first time, it gets called once. When I call it again, it gets called 2x. Third time, 4x, fourth 8x, etc.

* what you expected to happen
That it would be sent once.
The number of matches should not be undefined.
If there are 4 matches of the message (ex. "(!) Balance: $55,555.50" four times) it should set matches to 4.

In particular it will be useful to provide these informations :
* what mineflayer version?
- 6.5.0
* what server version?
- 1.8.8 (client is 1.8.8)
* which server? (spigot, paper, vanilla, any particular plugins ?)
- Spigot

**Comment by rom1504** [comment]
Why not simply use 'message' or 'chat' event and a regex ?

**Comment by GoogleSites** [comment]
I've changed my code to the following:

    Minecraft.chat('/bal');
    Minecraft.on('message', msg => {
            if(msg.toString()) {
                var regex = /^\(!\) Balance: \$?([0-9]{1,3},([0-9]{3},)*[0-9]{3}|[0-9]+)(.[0-9][0-9])?$/
                var balanceInput = msg.toString();
                var balanceOutput = balanceInput.match(regex);
                if(balanceInput != null) {
                    console.log(balanceOutput);
                }

When I run through it the first time, it finds it once, second time twice, third time 4x, five times 8x, etc.
Basically, it does the same thing.

**Comment by GoogleSites** [comment]
I'd like to note that it adds another match to the regex each time.
/bal outputs my balance (which looks something like "(!) Balance: $60,000.00").

1st time shows balance once in chat (1 output with regex)
2nd time shows balance twice in chat (4 outputs with regex)
3rd time shows balance three times in chat (8 outputs with regex)
etc.

I would assume that it compares the chat output for each message chat comes through to the client - so I'm not sure how it's outputting the matches so many times (it should only output the balance whenever it sees it come through the chat - which would be one at a time in this case).

**Comment by rom1504** [comment]
can I see the full code you're executing ?
I believe you're doing something broken that is not in the code you provided.

**Comment by rom1504** [comment]
Something obvious, just in case : you should execute the .on only one time, it listens to all instances of the "message" event.

**Comment by GoogleSites** [comment]
const Discord = require ("discord.js");
const config = require("../config.json");
const colour = config.colour;
const muterole = config.muteRole;
const staffrole = config.staffRole;

module.exports.run = async (bot, message, args) => {
message.delete();
    if(message.member.roles.find(r => r.name === staffrole)) {
        var member = message.mentions.members.first();
        let mute = message.guild.roles.find(x => x.name === muterole)
            if (!member.roles.has(mute.id)) {
            member.addRole(mute).catch(console.error);
            const embed = new Discord.RichEmbed()
                .setTitle("Code 200")
                .setColor(colour)
                .setDescription("Successfully muted" + member + ".")
                .setTimestamp()
                .setFooter(`${config.footer}`);
            message.channel.send(embed)
            .then(msg => {
                msg.delete(5000)
        const logChannel = message.guild.channels.find(c => c.name === config.mainLogs && c.type === 'text');
        let mutemessage = new Discord.RichEmbed()
            .setTitle("Mute")
            .setThumbnail(message.author.avatarURL)
            .setColor(colour)
            .addField("Punisher", message.author.tag, true)
            .addField("Player Muted", member.displayName + "\n\u200B", true)
            .setTimestamp()
            .setFooter(`${config.footer}`);
        logChannel.send(mutemessage)
        })
      }
  }
  return message.delete();
}

module.exports.config = {
  name: "mute",
  aliases: [""]
}

It runs when I type !bank in a Discord channel.

**Comment by GoogleSites** [comment]
const Discord = require("discord.js");
const config = require("../config.json");
const colour = config.colour;
const fs = require("fs");

module.exports.run = async (client, message, args, Minecraft) => {
    const thisChannel = message.channel;
    var finalBalance = null;
    
    message.delete();
    Minecraft.chat('/bal');
    Minecraft.on('message', msg => {
            if(msg.toString()) {
                var regex = /^\(!\) Balance: \$?([0-9]{1,3},([0-9]{3},)*[0-9]{3}|[0-9]+)(.[0-9][0-9])?$/
                var balanceInput = msg.toString();
                var balanceOutput = balanceInput.match(regex);
                if(balanceInput != null) {
                    console.log(balanceOutput);
                }
            }
    })
};
module.exports.config = {
  name: "bank",
  aliases: [""]
}

This all runs when I type !bank in a Discord channel - no errors anywhere.

**Comment by GoogleSites** [comment]
Okay wait I just realized I have it start the event every time I run the commands :/

**Comment by rom1504** [comment]
so this is fixed
